swatch-core/src/main/java/org/candlepin/subscriptions/db/EventRecordRepository.java

```diff
@@ -82,9 +83,11 @@ Stream<EventRecord> findByOrgIdAndTimestampGreaterThanEqualAndTimestampLessThanO
   /**
    * Delete old event records given a cutoff date
    *
-   * @param cutoffDate Dates BEFORE this timestamp get deleted
+   * @param cutoffDate Dates before this timestamp get deleted
    */
-  void deleteEventRecordsByTimestampBefore(OffsetDateTime cutoffDate);
+  @Modifying
+  @Query("DELETE FROM EventRecord e WHERE e.timestamp<:cutoffDate")
+  void deleteInBulkEventRecordsByTimestampBefore(OffsetDateTime cutoffDate);
```

Comment:
Would you mind adding to the javadoc why this method exists?  Just something like what you have in the PR description.

```
By default, a derived delete query will return instances matching the where clause prior to deleting the entities one at a time. This was too memory intensive.  Since we're not concerned with invoking any lifecycle callbacks, we can override the behavior with @query to delete all matching records in bulk.  See the <a href="https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.modifying-queries.derived-delete">Spring Data documentation</a> on deletion.
```

Reply:
i can do that